editor/src/core/shared/math-utils.ts

```diff
@@ -1109,3 +1109,12 @@ export function resize(
   const desiredHeight = (desiredSize.width / aspectRatio) ^ 0
   return size(desiredSize.width, desiredHeight)
 }
+
+export function intersection<T>(sets: Array<Set<T>>): Set<T> {
+  const [first, ...rest] = sets
+  if (first == null) {
+    return new Set()
+  }
+
+  return rest.reduce((acc, v) => new Set([...acc].filter((x) => v.has(x))), first)
+}
```

Comment:
We have a `set-utils.ts` that would be better suited for this function

Reply:
moved it there! (the function in https://github.com/concrete-utopia/utopia/pull/3270/commits/1d3501c17e198f945b8bc53a0ce541efab703bf9, the tests in https://github.com/concrete-utopia/utopia/pull/3270/commits/035a7ac98b157c4b1a4a8e4d1140719e9a8d05f9)